shared/js/views.jsx

```diff
@@ -911,73 +920,63 @@ loop.shared.views = (function(_, mozL10n) {
-      showTile: React.PropTypes.bool.isRequired,
       video: React.PropTypes.object.isRequired
     },
 
-    isLocalMediaAbsolutelyPositioned: function(matchMedia) {
-      if (!matchMedia) {
-        matchMedia = this.props.matchMedia;
-      }
-      return matchMedia &&
-        // The screen width is less than 640px and we are not screen sharing.
-        (matchMedia("screen and (max-width:640px)").matches ||
-         // or the screen width is less than 300px.
-         (matchMedia("screen and (max-width:300px)").matches));
+    _shouldShowRemoteStream: function() {
+      return this.props.remoteSrcMediaElement ||
+        this.props.remotePosterUrl || this.props.isRemoteLoading;
     },
 
-    getInitialState: function() {
-      return {
-        localMediaAboslutelyPositioned: this.isLocalMediaAbsolutelyPositioned()
-      };
-    },
-
-    componentWillReceiveProps: function(nextProps) {
-      // This is all for the ui-showcase's benefit.
-      if (this.props.matchMedia !== nextProps.matchMedia) {
-        this.updateLocalMediaState(null, nextProps.matchMedia);
-      }
-    },
-
-    componentDidMount: function() {
-      window.addEventListener("resize", this.updateLocalMediaState);
-    },
-
-    componentWillUnmount: function() {
-      window.removeEventListener("resize", this.updateLocalMediaState);
-    },
-
-    updateLocalMediaState: function(event, matchMedia) {
-      var newState = this.isLocalMediaAbsolutelyPositioned(matchMedia);
-      if (this.state.localMediaAboslutelyPositioned !== newState) {
-        this.setState({
-          localMediaAboslutelyPositioned: newState
-        });
-      }
-    },
+    render: function() {
+      var mediaLayoutClasses = classNames({
+        "media-layout": true,
+        "showing-local-streams": this.props.localSrcMediaElement ||
+          this.props.localPosterUrl,
+        "showing-media-wait": this.props.showMediaWait,
+        "showing-remote-streams": this._shouldShowRemoteStream()
+      });
 
-    renderLocalVideo: function() {
       return (
-        <div className="local">
+        <div className={mediaLayoutClasses}>
+          <MediaView
+            displayAvatar={!this.props.renderRemoteVideo}
+            isLoading={this.props.isRemoteLoading}
+            mediaType="remote"
+            posterUrl={this.props.remotePosterUrl}
+            srcMediaElement={this.props.remoteSrcMediaElement} />
           <MediaView
             displayAvatar={this.props.localVideoMuted}
             isLoading={this.props.isLocalLoading}
             mediaType="local"
             posterUrl={this.props.localPosterUrl}
             srcMediaElement={this.props.localSrcMediaElement} />
+          <MediaButtonsView
+            audio={this.props.audio}
+            dispatcher={this.props.dispatcher}
+            leaveRoom={this.props.leaveRoom}
+            video={this.props.video} />
+          <MediaWaitView
+            showMediaWait={this.props.showMediaWait} />
```

Comment:
I'm not seeing the media wait view on the standalone, when I'm being prompted for gUM. Did something break?